client/my-sites/patterns/index.tsx

```diff
@@ -0,0 +1,41 @@
+import { UniversalNavbarHeader } from '@automattic/wpcom-template-parts';
+import Main from 'calypso/components/main';
+import { PatternsCategoryPage } from 'calypso/my-sites/patterns/pages/category';
+import { PatternsHomePage } from 'calypso/my-sites/patterns/pages/home';
+import { useSelector } from 'calypso/state';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import type { PatternGalleryFC } from 'calypso/my-sites/patterns/types';
+
+import './style.scss';
+
+type PatternsIndexProps = {
+	category: string;
+	isGridView?: boolean;
+	patternGallery: PatternGalleryFC;
+};
+
+export const PatternsIndex = ( {
```

Comment:
Instead of adding this component, we could also implement the same logic in `renderPatterns` in `index.node` and `index.web` respectively. Do you think this approach is better?

Reply:
I think that this approach is better:
1) There is an answer above https://github.com/Automattic/wp-calypso/pull/88244#discussion_r1514999285
2) Also, with this approach we avoid duplications
3) With separate file we decouple - logic (index.node and index.web) from rendering (index)

Reply:
Fair enough! Could we maybe rename this to `PatternsWrapper` or something, though? `PatternsIndex ` sounds like this might be the home page for patterns

Reply:
>PatternsIndex  sounds like this might be the home page for patterns

Of course, it's personal, but it doesn't sound to me this way. And when I started my journey in development many years ago - exactly PHP examples showed me the difference - typically index.php was used as a layout - common header and footer plus some body (based on a route). So `index` for me is entry point, when `home` is mostly about `body` inside `index`.

ChatGPT says:
```
Differences
Functionality Scope: index.php is often used for initialization and routing, wrapper.php for shared functionality or layout wrapping, and home.php for defining the homepage content.
```

So maybe we could use `wrapper`, but exactly our case looks like "typical index.php" file to me :) 

WDYT?

Reply:
`index` to me is the default lookup location in a folder, i.e., the home page. [This Wikipedia article](https://en.wikipedia.org/wiki/Web_server_directory_index) puts it well:
> When an HTTP client (generally a web browser) requests a URL that points to a directory structure instead of an actual web page within the directory structure, the web server will generally serve a default page, which is often referred to as a main or "index" page.

This is also true for import paths in Calypso, which is why `client/my-sites/patterns/components/header/index.tsx` is looked up when we call `import { PatternsHeader } from 'calypso/my-sites/patterns/components/header'`, despite there not being an `index.tsx` in the import path.

Reply:
👍 Renamed